### ISSUE TITLE
Error on the cluster with runSCENIC_2_createRegulons

### ISSUE THREAD

**Issue by selcant** [aertslab/SCENIC#35]
Hi,
I am trying to run the SCENIC pipeline on a high performance cluster and keep getting the same error at "RcisTarget: Adding motif annotation" step. The error is: 
`Error in serialize(data, node$con, xdr = FALSE) : 
  error writing to connection
Calls: runSCENIC_2_createRegulons ... .send_EXEC -> <Anonymous> -> sendData.SOCK0node -> serialize`
I tried running with various number of cores and memory allocations on a single node but keep getting the same error. Any suggestions on settings to use on the cluster to avoid this issue? 
Thank you!

**Comment by s-aibar** [comment]
Hello,

The function `runSCENIC_2_createRegulons()` uses `BiocParallel::bplapply`. It seems that the default settings are not addequate on some systems, so you probably need to adjust the settings for BiocParallel. You can do some tests with this code:
```
library(BiocParallel)
registered() # your current settings
system.time(bplapply(1:1000, sqrt))

register(MulticoreParam(workers = 10), default = TRUE)
# or: register(SerialParam(), default = TRUE)
system.time(bplapply(1:1000, sqrt))
```

Plesase, let me know whether this helps. I will also add a comment to the tutorial to clarify this. Thank you for your feedback!
Sara

**Comment by selcant** [comment]
Thank you! Sorry, I forgot to update here. Yes, I used register to fix it. I added this to my code:
`library(BiocParallel)
register(MulticoreParam(workers=cores), default = TRUE)
register(SnowParam(workers=cores), default = TRUE)
`
So I can pass along the number of cores I want the script to use. Thank you for the reply! 

I do have a separate question about the motif databases. I keep getting inconsistent results with the same seed, more specifically one TF (Sox2) is missing in the new runs I do which was present earlier. I had re-downloaded the cisTarget databases, so that is the only difference. Do they get updated or modified? I am using the same files: mm9-500bp-upstream-7species.mc9nr and mm9-tss-centered-10kb-7species.mc9nr.

Thanks again for the reply!
